go/obscuronode/enclave/enclave.go

```diff
@@ -391,6 +380,48 @@ func (e *enclaveImpl) IsInitialised() bool {
 	return e.storage.FetchSecret() != nil
 }
 
+func (e *enclaveImpl) isGenesisBlock(block *types.Block) bool {
+	return e.l1Blockchain != nil && block.Hash() != e.l1Blockchain.Genesis().Hash()
+}
+
+// Inserts the block into the L1 chain if it exists and the block is not the genesis block. Returns a non-nil
+// BlockSubmissionResponse if the insertion failed.
+func (e *enclaveImpl) insertBlockIntoL1Chain(block *types.Block) *nodecommon.BlockSubmissionResponse {
+	if e.l1Blockchain != nil {
```

Comment:
wouldn't it be nicer to do:
```
if... == nil{
   return nil
}
....
```

Reply:
If you do that, you have to leave in an extra `return nil` at the end, for the case where the insertion happens but doesn't cause an error. So it's not really nicer IMO.